sources/test/test_curl_source.py

```diff
@@ -80,3 +80,24 @@ def test_curl_source_amend_secrets_fail(sources_module):
     with pytest.raises(RuntimeError) as exc:
         curl_source.amend_secrets(checksum, desc)
     assert "mtls secrets required" in str(exc)
+
+
+class FakeSubscriptionManager:
+    def get_secrets(self, url):
+        return f"secret-for-{url}"
+
+
+def test_curl_source_amend_secrets_subscription_mgr(sources_module):
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    curl_source = sources_module.CurlSource.from_args(["--service-fd", str(sock.fileno())])
+    desc = {
+        "url": "http://localhost:80/a",
+        "secrets": {
+            "name": "org.osbuild.rhsm",
+        },
+    }
+
+    curl_source.subscriptions = FakeSubscriptionManager()
+    checksum = "sha256:1234567890123456789012345678901234567890909b14ffb032aa20fa23d9ad6"
```

Comment:
I remember you mentioned issue with hardcoded strings in case of parallel test run. If we need to think about this here? Shall `checksum` be randomly generated (at least partially)?

Reply:
Thanks for your review! The issue we discussed earlier was trying to avoid using fixed UUIDs (c.f. https://github.com/osbuild/osbuild/pull/1641) in tests. As UUIDs supposed to be globally unique. However here it's just a checksum so this guideline does not apply. There is a bit of an open question if this checksum shoiuld actually look more like 11111... to make sure that it's clear that it's not actually meaningful. I can fix that :)

Reply:
Thanks, solved for me :)